https://github.com/supertuxkart/stk-code/commit/5e05f1178ce6bc5f3a653b55ab3dc6d016196341.patch

From 5e05f1178ce6bc5f3a653b55ab3dc6d016196341 Mon Sep 17 00:00:00 2001
From: Deve <deveee@gmail.com>
Date: Mon, 3 Oct 2016 23:26:09 +0200
Subject: [PATCH] Fixed compiler error on Linux with non-x86 64bit platforms,
 e.g. arm64, mips, and s390x architectures

This modification is already applied in upstream angelscript repository:
https://sourceforge.net/p/angelscript/code/2353/

Thanks to Adrian Bunk and Andreas Jonsson
---
 lib/angelscript/projects/cmake/CMakeLists.txt | 1 +
 lib/angelscript/source/as_config.h            | 2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/lib/angelscript/projects/cmake/CMakeLists.txt b/lib/angelscript/projects/cmake/CMakeLists.txt
index e93971315e..755d8378c3 100644
--- a/lib/angelscript/projects/cmake/CMakeLists.txt
+++ b/lib/angelscript/projects/cmake/CMakeLists.txt
@@ -67,6 +67,7 @@ set(ANGELSCRIPT_SOURCE
     ../../source/as_builder.cpp
     ../../source/as_bytecode.cpp
     ../../source/as_callfunc.cpp
+    ../../source/as_callfunc_mips.cpp
     ../../source/as_callfunc_x86.cpp
     ../../source/as_callfunc_x64_gcc.cpp
     ../../source/as_callfunc_x64_msvc.cpp
diff --git a/lib/angelscript/source/as_config.h b/lib/angelscript/source/as_config.h
index cb05bffbd5..5bb5b8e800 100644
--- a/lib/angelscript/source/as_config.h
+++ b/lib/angelscript/source/as_config.h
@@ -844,7 +844,7 @@
 			#define THISCALL_PASS_OBJECT_POINTER_ON_THE_STACK
 			#define AS_X86
 			#undef AS_NO_THISCALL_FUNCTOR_METHOD
-		#elif defined(__LP64__) && !defined(__arm64__)
+		#elif defined(__x86_64__)
 			#define AS_X64_GCC
 			#undef AS_NO_THISCALL_FUNCTOR_METHOD
 			#define HAS_128_BIT_PRIMITIVES
